.github/workflows/docbuild.yml

```diff
@@ -2,6 +2,9 @@ name: docbuild
 
 on: [push, pull_request]
 
+env:
+  UV_SYSTEM_PYTHON: 1
```

Comment:
Great PR @dangotbanned! Do you have a reference why this is environment definition is necessary? Thanks!

Reply:
Thanks @mattijn!
Yeah I can explain.

I've enabled [`UV_SYSTEM_PYTHON`](https://docs.astral.sh/uv/configuration/environment/#uv_system_python) as a [workflow environment variable](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#defining-environment-variables-for-a-single-workflow) to add the `--system` argument on all relevant `uv` commands.
You can see in [`narwhals/.github/workflows/downstream_tests.yml`](https://github.com/narwhals-dev/narwhals/blob/7316b2eaa35801559876f42bc911d1285f28c3ef/.github/workflows/downstream_tests.yml) that argument is [used **25** times](https://github.com/search?q=repo%3Anarwhals-dev%2Fnarwhals%20path%3A.github%2Fworkflows%2Fdownstream_tests.yml%20%22--system%22&type=code). 

So this is just to keep it simple, making sure it is there when needed - but isn't something you need to worry about if updating the workflow in the future 🙂

Reply:
I really need to transition away from windows at some point 😔

Think it is causing problems in https://github.com/vega/vega-datasets/pull/648#issuecomment-2558086569 as well